frontend/src/app/containers/ExperimentPage/ExperimentControls.js

```diff
@@ -1,76 +1,63 @@
 import React from "react";
-import classnames from "classnames";
 import { Localized } from "fluent-react/compat";
 import LocalizedHtml from "../../components/LocalizedHtml";
 import { experimentL10nId, isMobile } from "../../lib/utils";
+import {
+  FeedbackButton,
+  MobileTriggerButton,
+  GraduatedButton,
+  UninstallButton,
+  InstallTestPilotButton,
+  EnableExperimentButton,
+  MobileStoreButton,
+  WebExperimentButton
+} from "./ExperimentButtons";
 
 import type {
   ExperimentControlsType,
-  WebExperimentControlsType,
-  EnableButtonType
+  CreateButtonsType
 } from "./types";
 
 export default function ExperimentControls({
-  isMinFirefox,
-  hasAddon,
-  userAgent,
+  doShowEolDialog,
+  doShowMobileAppDialog,
+  doShowPreFeedbackDialog,
+  enabled,
   experiment,
-  installed,
+  flashMeasurementPanel,
   graduated,
-  enabled,
+  hasAddon,
   installExperiment,
-  uninstallExperimentWithSurvey,
-  isEnabling,
   isDisabling,
-  doShowEolDialog,
-  doShowPreFeedbackDialog,
-  flashMeasurementPanel,
+  isEnabling,
+  isMinFirefox,
   sendToGA,
   surveyURL,
-  doShowMobileAppDialog
+  uninstallExperimentWithSurvey,
+  userAgent
 }: ExperimentControlsType) {
   const {
-    title,
-    min_release,
     max_release,
+    min_release,
     platforms,
-    pre_feedback_copy,
-    slug
+    title
   } = experiment;
 
   const validVersion = isValidVersion(userAgent, min_release, max_release);
 
-  const handleFeedback = evt => {
-    if (pre_feedback_copy === null || !pre_feedback_copy) {
-      sendToGA("event", {
-        eventCategory: "ExperimentDetailsPage Interactions",
-        eventAction: "Give Feedback",
-        eventLabel: title,
-        dimension11: slug
-      });
-    } else {
-      doShowPreFeedbackDialog(evt);
-      sendToGA("event", {
-        eventCategory: "ExperimentDetailsPage Interactions",
-        eventAction: "Give Feedback",
-        eventLabel: experiment.title,
-        dimension11: slug
-      });
-    }
-  };
-
   const highlightPrivacy = () => {
-    document.querySelectorAll(".measurements").forEach(
-      el => {
-        if (el.offsetTop) {
-          window.scrollTo(0, el.offsetTop);
-        }
-      });
+    document.querySelectorAll(".measurements").forEach(el => {
+      if (el.offsetTop) {
+        window.scrollTo(0, el.offsetTop);
+      }
+    });
     flashMeasurementPanel();
   };
 
-  const buttons = platforms.map(platform => createButton({
+  const buttons = createButtons({
```

Comment:
No longer creating an array here, all the return possibility are generated in a switch statement in the `createButtons` function